apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx

```diff
@@ -91,48 +105,88 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
     const [ mappingChecked, setMappingChecked ] = useState<boolean>(false);
 
-    const nameField = useRef<HTMLElement>(null);
-    const regExField = useRef<HTMLElement>(null);
-    const displayOrderField = useRef<HTMLElement>(null);
-    const descriptionField = useRef<HTMLElement>(null);
+    const nameField: MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
+    const regExField: MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
+    const displayOrderField: MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
+    const descriptionField: MutableRefObject<HTMLElement> = useRef<HTMLElement>(null);
 
     const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
     const featureConfig: FeatureConfigInterface = useSelector((state: AppState) => state.config.ui.features);
     const [ hideSpecialClaims, setHideSpecialClaims ] = useState<boolean>(true);
+    const [ usernameConfig, setUsernameConfig ] = useState<ValidationFormInterface>(undefined);
 
     const { t } = useTranslation();
 
+    const { data: validationData } = useValidationConfigData();
+ 
+    /**
+     * Get username configuration.
+     */
+    useEffect(() => {
+        if (validationData) {
+            setUsernameConfig(getUsernameConfiguration(validationData));
+        }
+
+    }, [ validationData ]);
+ 
+    /**
+     * Get username configuration.
+     */
+    useEffect(() => {
+
+        if (
+            usernameConfig?.enableValidator === "true"
+            && attributeConfig?.systemClaims.includes(ClaimManagementConstants.EMAIL_CLAIM_URI)
+        ){
+            const emailClaimIndex: number 
+                = attributeConfig?.systemClaims.indexOf(ClaimManagementConstants.EMAIL_CLAIM_URI);
+
+            attributeConfig?.systemClaims.splice(emailClaimIndex, 1);
+        } else 
+        {
+            attributeConfig?.systemClaims.push(ClaimManagementConstants.EMAIL_CLAIM_URI);
+        }
+    }, [ usernameConfig, attributeConfig ]);
+
     useEffect(() => {
         if (claim?.supportedByDefault) {
             setIsShowDisplayOrder(true);
         }
         if (claim?.readOnly) {
             setIsClaimReadOnly(true);
         }
-        if (claim && (attributeConfig?.systemClaims.length <= 0
-            || attributeConfig?.systemClaims.indexOf(claim?.claimURI) === -1)) {
+        if (claim 
+            && (
+                attributeConfig?.systemClaims.length <= 0
+                || attributeConfig?.systemClaims.indexOf(claim?.claimURI) === -1
+            )
+        ) {
             setHideSpecialClaims(false);
         }
-    }, [ claim ]);
+        else {
+            setHideSpecialClaims(true);
+        }
+    }, [ claim, usernameConfig ]);
 
     useEffect(() => {
-        const dialectID = getDialectID();
+        const dialectID: string[] = getDialectID();
 
         if(claim) {
-            const externalClaimRequest = [];
+            const externalClaimRequest: any = [];
```

Comment:
Shall we avoid using `any` here?

Reply:
Addressed with 05049aa1cc24d53927699fa67f76598311183643.